### PR TITLE
Stronger Augmentation: 2x gap/stagger and DSDF sigma

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

New augmentation channels fail (input noise, Re noise, manifold mixup). But the EXISTING augmentations are proven effective — they're part of the baseline that beat the 16-seed ensemble. The question: are the current augmentation strengths OPTIMAL, or just "good enough"?

Current: `--aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05`
Proposed: `--aug_gap_stagger_sigma 0.04 --aug_dsdf2_sigma 0.10` (2x both)

**Why stronger augmentation might help:**
1. **More diverse virtual training examples.** 2x sigma means the training distribution covers a wider range of gap/stagger/DSDF perturbations, creating more virtual configurations for the model to learn from.
2. **Directly targets OOD.** Gap and stagger perturbations create virtual tandem configurations near the OOD boundary. Stronger perturbations push the virtual samples further toward OOD territory, potentially improving p_oodc and p_tan.
3. **Proven augmentation channels.** Unlike input noise (which corrupted geometric features) or Re noise (which disrupted critical conditioning), gap/stagger and DSDF augmentations are physics-motivated perturbations that have been validated in the baseline.

**Risk:** If 0.02/0.05 was already at the sweet spot, 2x may push past it — creating augmented samples that are too far from the real data distribution, degrading training signal quality.

**Confidence:** Medium. This is a hyperparameter sweep on proven augmentation channels. The outcome depends entirely on whether the current values are at the optimum or below it.

## Instructions

**No code changes needed.** Change two flag values:

### Run 2 seeds

```bash
# Seed 42
cd cfd_tandemfoil && python train.py \
  --agent thorfinn --wandb_name "thorfinn/strong-aug-s42" \
  --wandb_group "round19/stronger-augmentation" \
  --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.04 --aug_dsdf2_sigma 0.10 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature

# Seed 73 — identical but --seed 73 --wandb_name "thorfinn/strong-aug-s73"
```

Changes: `--aug_gap_stagger_sigma 0.04` (was 0.02), `--aug_dsdf2_sigma 0.10` (was 0.05)

### Report results

Table: p_in, p_oodc, p_tan, p_re for both seeds, 2-seed avg, baseline comparison, W&B run IDs.

Focus on p_tan and p_oodc — tandem-specific augmentation should most affect these.

## Baseline

| Metric | Baseline | Target to beat |
|--------|----------|----------------|
| p_in   | **11.979** | < 11.98 |
| p_oodc | **7.643**  | < 7.65  |
| p_tan  | **28.341** | < 28.34 |
| p_re   | **6.300**  | < 6.30  |

W&B: `hgml7i2r` (s42), `qic03vrg` (s73)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent thorfinn --wandb_name "thorfinn/baseline" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT thorfinn: Previous runs OOM'd / were cut short — both processes were assigned to the same GPU (0), causing s42 to crash with OOM at epoch 1. Relaunched both with explicit GPU pinning (s42 on GPU 0, s73 on GPU 1) and `PYTORCH_ALLOC_CONF=expandable_segments:True`. Both are now running fresh. W&B runs: s42=g396u0m2 (new run starting), s73=hwhy6z9e (new run starting). Will report results when complete.

**Comment by tcapelle** [comment]
STUDENT thorfinn:

## Results

### Stronger Augmentation: 2x gap/stagger and DSDF sigma

| Metric | Baseline | s42 | s73 | 2-seed Avg | Delta |
|--------|----------|-----|-----|------------|-------|
| p_in   | **11.979** | 12.268 | 12.004 | 12.136 | +1.3% ❌ |
| p_oodc | **7.643**  | 7.835  | 7.341  | 7.588  | -0.7% ✅ |
| p_tan  | **28.341** | 28.464 | 28.371 | 28.418 | +0.3% ❌ |
| p_re   | **6.300**  | 6.336  | 6.474  | 6.405  | +1.7% ❌ |

**W&B runs:** `b1g3lrs0` (s42), `h3hmt6ov` (s73)
**Epochs:** 149 each (both hit the epoch/timeout limit)
**Best epoch:** 148 for both seeds

**Commands used:**
```bash
# s42 (GPU 0)
CUDA_VISIBLE_DEVICES=0 PYTORCH_ALLOC_CONF=expandable_segments:True python train.py \
  --agent thorfinn --wandb_name "thorfinn/strong-aug-s42" \
  --wandb_group "round19/stronger-augmentation" --seed 42 \
  --aug_gap_stagger_sigma 0.04 --aug_dsdf2_sigma 0.10 \
  [... all other flags identical to baseline ...]

# s73 (GPU 1) — identical but --seed 73 --wandb_name "thorfinn/strong-aug-s73"
```

### What happened

**Negative result.** Doubling the augmentation sigmas (gap/stagger 0.02→0.04, DSDF 0.05→0.10) hurt 3 of 4 metrics:

- **p_in (+1.3%):** The stronger perturbations created augmented samples too far from the in-distribution data, degrading training signal quality. This is the clearest regression.
- **p_oodc (-0.7%):** The one bright spot — stronger augmentation did help generalization to OOD conditions, but only modestly and only on average (s73 helped, s42 hurt).
- **p_tan (+0.3%):** Despite the hypothesis that stronger gap/stagger augmentation would help tandem predictions, it barely moved. The tandem transfer task appears insensitive to augmentation strength at these levels.
- **p_re (+1.7%):** Worst regression. Re perturbation isn't directly augmented here, so this suggests the training quality degradation from noisier augmentation had a spillover effect.

**Conclusion:** The current augmentation strengths (0.02/0.05) appear to be at or near the optimum. The hypothesis that they were "below the sweet spot" is rejected — 2x sigma pushes past the sweet spot, creating augmented samples that are too far from the real distribution. The risk noted in the hypothesis (degraded training signal) materialized.

### Suggested follow-ups

- **Intermediate values** (e.g., 0.03/0.075 — 1.5x instead of 2x) could be worth testing if there's interest, though the marginal gains would likely be small given how close we already are to optimal.
- **Asymmetric scaling** — the small p_oodc improvement with stronger augmentation suggests gap/stagger could potentially be increased slightly while keeping DSDF at 0.05, but the effect is within noise.
- The baseline augmentation strengths appear well-tuned. Future augmentation work should likely focus on new augmentation channels rather than rescaling existing ones.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2248 because ADVISOR: Closing — 3/4 metrics regressed (p_in +1.3%, p_tan +0.3%, p_re +1.7%). The small p_oodc improvement (-0.7%) is within seed noise. Current augmentation sigmas (0.02/0.05) are at or near the sweet spot — doubling them pushes augmented samples too far from the real distribution. Augmentation strength is now a confirmed optimized parameter. Good analysis on asymmetric scaling, but the effect is too marginal to pursue..
